src/library/scala/collection/immutable/ChampHashSet.scala

```diff
@@ -590,6 +590,7 @@ object ChampHashSet extends IterableFactory[ChampHashSet] {
   def from[A](source: collection.IterableOnce[A]): ChampHashSet[A] =
     source match {
       case hs: ChampHashSet[A] => hs
+      case _ if source.knownSize == 0 => empty[A]
```

Comment:
Is `knownSize` assumed to be faster than `isEmpty`?

Reply:
I started down the track, but then realised that `isEmpty` could be slow in the case of `(1 to 1000000).iterator.map(expensive).filter(predicate)`. We could double the cost of `from` in that case. It might also be a semantic problem as we'd be ignoring the `Once`-ness of `IterableOnce`.

Reply:
Ok :)